src/psyclone/tests/domain/lfric/transformations/raise_psyir_2_lfric_kern_trans_test.py

```diff
@@ -0,0 +1,271 @@
+routine to PSyclone kernel-layer-specific PSyIR which uses specialised
+classes.
+
+'''
+import pytest
+
+from psyclone.domain.lfric.kernel.psyir import LFRicContainer
+from psyclone.domain.lfric.transformations.raise_psyir_2_lfric_kern_trans \
+    import RaisePSyIR2LFRicKernTrans
+from psyclone.domain.gocean.transformations.raise_psyir_2_gocean_kern_trans \
+    import find_symbol
+from psyclone.psyir.nodes import FileContainer, Container, Routine
+from psyclone.psyir.symbols import SymbolTable, DataTypeSymbol
+from psyclone.psyir.transformations import TransformationError
+
+METADATA = (
+    "type, extends(kernel_type) :: testkern_type\n"
+    "   type(arg_type), dimension(1) :: meta_args =  (/     &\n"
+    "           arg_type(gh_field,    gh_real, gh_inc,  w1) &\n"
+    "         /)\n"
+    "   integer :: operates_on = cell_column\n"
+    " contains\n"
+    "   procedure, nopass :: code => testkern_code\n"
+    "end type testkern_type\n")
+
+
+PROGRAM = (
+    f"module dummy\n"
+    f"{METADATA}"
+    f"contains\n"
+    f"  subroutine compute_cu_code()\n"
+    f"  end subroutine compute_cu_code\n"
+    f"end module dummy\n")
+
+
+# find_symbol function
+
+def test_find_symbol(fortran_reader):
+    '''Test that the PSyIR tree walk to find the metadata symbol works if
+    multiple symbol tables need to be searched before finding the
+    correct one.
+
+    '''
+    my_program = (f"module dummy\n"
+                  f"contains\n"
+                  f"subroutine dummy1()\n"
+                  f"end subroutine\n"
+                  f"  subroutine dummy2()\n"
+                  f"{METADATA}"
+                  f"  end subroutine\n"
+                  f"end module\n")
+    kernel_psyir = fortran_reader.psyir_from_source(my_program)
+    symbol, node = find_symbol(kernel_psyir, "testkern_type")
+    assert isinstance(symbol, DataTypeSymbol)
+    assert symbol.name == "testkern_type"
+    assert isinstance(node, Routine)
+    assert node.name == "dummy2"
+    symbol, node = find_symbol(kernel_psyir, "not_here")
+    assert symbol is None
+    assert node is None
+
+
+def test_kerntrans_init():
+    '''Test that an instance of RaisePSyIR2LFRicKernTrans can be
+    succesfully created and raises the expected exception if the
+    supplied argument is invalid.
+
+    '''
+    kern_trans = RaisePSyIR2LFRicKernTrans()
+    assert isinstance(kern_trans, RaisePSyIR2LFRicKernTrans)
+
+
+def test_validate_nosymbol(fortran_reader):
+    '''Test that the validate method raises an exception if the
+    metadata_name property does not correspond to a symbol in the
+    supplied PSyIR.
+
+    '''
+    kernel_psyir = fortran_reader.psyir_from_source(PROGRAM)
+    kern_trans = RaisePSyIR2LFRicKernTrans()
+    with pytest.raises(TransformationError) as info:
+        kern_trans.validate(kernel_psyir, {"metadata_name": "does_not_exist"})
+    assert ("The metadata name (does_not_exist) provided to the "
```

Comment:
I think quotes around the supplied name (rather than parentheses) would be clearer and more consistent with our usual practice.

Reply:
Agreed. Not sure what I was thinking there.